servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -15,351 +15,56 @@
-    String masterHostname =
-        tachyonConf.get(Constants.MASTER_HOSTNAME, NetworkUtils.getLocalHostName(tachyonConf));
-    int masterPort = tachyonConf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
-    String workerHostName = NetworkUtils.getLocalHostName(tachyonConf);
-    int workerPort = tachyonConf.getInt(Constants.WORKER_PORT, Constants.DEFAULT_WORKER_PORT);
-    int dataPort =
-        tachyonConf.getInt(Constants.WORKER_DATA_PORT, Constants.DEFAULT_WORKER_DATA_SERVER_PORT);
-    int minWorkerThreads =
-        tachyonConf.getInt(Constants.WORKER_MIN_WORKER_THREADS, Runtime.getRuntime()
-            .availableProcessors());
-    int maxWorkerThreads =
-        tachyonConf.getInt(Constants.WORKER_MAX_WORKER_THREADS,
-            Constants.DEFAULT_WORKER_MAX_WORKER_THREADS);
-
-    return new TachyonWorker(new InetSocketAddress(masterHostname, masterPort),
-        new InetSocketAddress(workerHostName, workerPort), dataPort, minWorkerThreads,
-        maxWorkerThreads, tachyonConf);
-  }
-
-  private static void setMasterAddress(String masterAddress, TachyonConf conf) {
-    if (masterAddress == null) {
-      return;
-    }
-    String masterHostnameConf =
-        conf.get(Constants.MASTER_HOSTNAME, NetworkUtils.getLocalHostName(conf));
-    String masterPortConf = conf.get(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT + "");
-    String[] address = masterAddress.split(":");
-    String masterHostname = address[0];
-    if (!masterHostnameConf.equals(masterHostname)) {
-      LOG.warn("Master host in configuration ({}) is different from the command line ({}).",
-          masterHostnameConf, masterHostname);
-      conf.set(Constants.MASTER_HOSTNAME, masterHostname);
-    }
-    String masterPort = masterPortConf;
-    if (address.length > 1) {
-      masterPort = address[1];
-      if (!masterPortConf.equals(masterPort)) {
-        LOG.warn("Master port in configuration ({}) is different from the command line ({}).",
-            masterPortConf, masterPort);
-        conf.set(Constants.MASTER_PORT, masterPort);
-      }
-    }
-    return;
-  }
-
-  public static void main(String[] args) throws UnknownHostException {
-    if (args.length > 1) {
-      LOG.info("Usage: java -cp target/tachyon-" + Version.VERSION + "-jar-with-dependencies.jar "
-          + "tachyon.Worker [<MasterHost:Port>]");
-      System.exit(-1);
-    }
-
+  public static void main(String[] args) {
+    checkArgs(args);
     TachyonConf tachyonConf = new TachyonConf();
-    if (args.length == 1) {
-      setMasterAddress(args[0], tachyonConf);
-    }
-
-    String resolvedWorkerHost = NetworkUtils.getLocalHostName(tachyonConf);
-    LOG.info("Resolved local TachyonWorker host to " + resolvedWorkerHost);
-
+    BlockWorker worker = null;
     try {
-      TachyonWorker worker = TachyonWorker.createWorker(tachyonConf);
-      worker.start();
+      worker = new BlockWorker(tachyonConf);
```

Comment:
NIT: Probably treat `worker = new BlockWorker(tachyonConf);` and `worker.process();` in different try-catch blocks, and `LOG.error` differently (after all, if worker can not be initialized, there is no "shutting down worker" at all)? And then you don't need to check if `worker == null`